cityofaustin/atd-moped
# Fix bug where clicking a drawn line selects the underlying street segment as well

## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/14569

This PR fixes an issue where clicking on drawn lines or point when creating or editing a component also selected lines or points in the selectable layers they overlapped. When a click event happens, we can access `e.features` to see an array of all features captured in the click in all layers below the mouse. 

The original code only checked the first feature in this array, and the fix was to check if any of the features were drawn before interacting with the selectable CTN layers. This paired with disabling interactive map layers while drawing fixed this up.

Heads up - I found an edge case and captured it in https://github.com/cityofaustin/atd-data-tech/issues/14855

## Testing
**URL to test:** 

https://deploy-preview-1219--atd-moped-main.netlify.app/moped/

**Steps to test:**
1. Go to a new or existing project
2. Create a new line component
   - Click a line from the dotted CTN layer to select it
   - Start drawing a line component and be sure to end it right on top of the line selected in the last step
   - You should see that the draw click no longer deselects the line from the first step
   - Click a spot on the map away from dotted lines and your features
   - Click on your drawn line where it intersects with your selected line. The drawn feature should highlight and the selected line below should remain selected
   - Save your component so you can edit it later
   - You can test this in staging with the same steps to see the bug in action
3. Try similar steps while editing the component that you just saved. You should see the drawn features win every time they are selected and are overlapping selected or unselected CTN layers - except for the edge case that I linked above.

---
#### Ship list
